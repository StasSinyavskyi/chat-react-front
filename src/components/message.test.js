/* eslint-env jest */
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import Message from './message';

jest.mock('./avatar', () => () => 'Avatar');

// if we have several test we can join them in describe
describe('<Message />', () => {
  const testParam = (
    <Message
      activeUser={{ _id: '123' }}
      sender={{ _id: '123', firstName: 'Test', lastName: 'Test' }}
      content="Test message"
      createdAt=""
      statusMessage={false}
    />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly', () => {
    const tree = renderer
      .create(testParam)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  // test 2

  const testParam2 = (
    <Message
      activeUser={{ _id: '123' }}
      sender={{ _id: '123', firstName: 'Test', lastName: 'Test' }}
      content="Test message"
      createdAt=""
      statusMessage
    />
  );

  it('renders without crashing 2', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam2, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly 2', () => {
    const tree = renderer
      .create(testParam2)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
