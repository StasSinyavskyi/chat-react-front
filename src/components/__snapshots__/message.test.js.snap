// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> render correctly 1`] = `
<div
  className="Message-messageWrapper-58 Message-messageWrappperFromMe-59"
>
  <div
    className="MuiPaper-root-64 MuiPaper-shadow2-68 MuiPaper-rounded-65 Message-message-60 Message-messageFromMe-61"
  >
    <span
      className="MuiTypography-root-91 MuiTypography-caption-101"
      style={
        Object {
          "color": "#8bc34a",
        }
      }
    >
      Test Test
    </span>
    <p
      className="MuiTypography-root-91 MuiTypography-body1-100"
    >
      Test message
    </p>
    <span
      className="MuiTypography-root-91 MuiTypography-caption-101"
    >
      Invalid date
    </span>
  </div>
  Avatar
</div>
`;

exports[`<Message /> render correctly 2 1`] = `
<div
  className="Message-messageWrapper-58"
>
  <p
    className="MuiTypography-root-91 MuiTypography-body1-100 Message-statusMessage-62"
  >
    <span
      className="MuiTypography-root-91 MuiTypography-caption-101 Message-statusMessageUser-63"
      style={
        Object {
          "color": "#8bc34a",
        }
      }
    >
      Test Test
    </span>
    Test message
    <span
      className="MuiTypography-root-91 MuiTypography-caption-101"
    >
      Invalid date
    </span>
  </p>
</div>
`;
